### PR TITLE
PLANET-4820: P4CG - Footer color should not be applied when "Main website navigation" is selected

### PR DESCRIPTION
**Pull request by pablocubico**
Note: despite of the issue description, I'm fixing other defaults and rules in this PR.

Ref: https://jira.greenpeace.org/browse/PLANET-4820

Styleguide PR: https://github.com/greenpeace/planet4-styleguide/pull/42
Blocks PR: https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/247

### CODE REVIEW

File: templates/blocks/site_logo.twig [hunk 1]
```diff
@@ -4,7 +4,8 @@
 	{% if logo == 'greenpeace' %}
 		<img src="{{ data_nav_bar.images }}gp-logo-{{ logo_color }}.svg" alt="Greenpeace">
 	{% else %}
-		<img src="{{ data_nav_bar.images }}{{ logo }}/logo-{{ logo_color }}.png" alt="Greenpeace">
+		{% set color = logo_color == 'green' ? 'light' : logo_color %}
```
Comment (pablocubico) [inline templates/blocks/site_logo.twig]: This is caused by another issue, it was added as a separate commit so we can revert it when the issue is solved.

Comment (pablocubico) [reply templates/blocks/site_logo.twig:7]: Details in the follow up ticket: https://jira.greenpeace.org/browse/PLANET-4901

**Comment by comzeradd** [comment]
Checking the branch locally I get some warnings when viewing a campaign that was already there

```
Notice: Undefined index: campaign_nav_type in single-campaign.php on line 79
Notice: Undefined index: campaign_nav_type in class-p4-post-campaign.php on line 494
Notice: Undefined index: campaign_nav_color in class-p4-post-campaign.php on line 516
Notice: Undefined index: campaign_nav_type in class-p4-post-campaign.php on line 521
Notice: Undefined index: campaign_nav_color in class-p4-post-campaign.php on line 535
```

File: classes/class-p4-post-campaign.php [hunk 2]
```diff
@@ -392,13 +394,14 @@ public static function css_vars( array $meta ): array {
 			$defaults = self::get_theme_defaults( $theme_json );
 
 			// Replace the defaults with the campaign options where applicable.
-			$intersect = array_intersect_key( $defaults, $meta );
+			$intersect = array_intersect_key( array_filter( $meta ), $defaults );
```
Comment (Inwerpsel) [inline classes/class-p4-post-campaign.php:397]: This line is to ensure that we only use keys from the meta that occur in the defaults, right? Maybe you can add a short comment like `// Use only meta keys that exist in the defaults` to make this a bit? Or maybe change the naming of the variable to something like `$metas_for_theme` to make this a bit more obvious?

File: classes/class-p4-post-campaign.php [hunk 3]
```diff
@@ -427,10 +428,11 @@ public static function get_body_font( $meta ): array {
 					'plastic'   => 'Montserrat',
 				];
 				$theme              = self::get_theme( $meta );
-				$body_font          = $campaigns_font_map[ $theme ];
+
+				$css_vars['campaign_body_font'] = $campaigns_font_map[ $theme ];
 			}
 
-			return [ 'campaign_body_font' => $body_font ];
+			return $css_vars;
```
Comment (Inwerpsel) [inline classes/class-p4-post-campaign.php:435]: If the `$css_vars` is passed by reference then this function doesn't need to return it as it's not used. Though personally I prefer not using passing by reference, as in function calls you can't really see that that is happening, and assigning the return value makes it a bit more obvious what is happening, though the extra assignment does make the syntax a bit more verbose.

File: classes/class-p4-post-campaign.php (hunk 2 above)
Comment (pablocubico) [reply classes/class-p4-post-campaign.php:397]: 👍  I just shamelessly copy-pasted your comment...

File: classes/class-p4-post-campaign.php (hunk 3 above)
Comment (pablocubico) [reply classes/class-p4-post-campaign.php:436]: Good point, I'll revert to the assignments.

**Pull request merged**